### DISCUSSION TITLE
New datetime no longer ISO - Discussion on changing this back

### DISCUSSION THREAD

**Discussion by alex-astronomer** [apache/airflow#24855] [Polls]
New:
![image](https://user-images.githubusercontent.com/89415310/177424921-9703e2e2-5f17-4a6f-ab86-2763bb534dae.png)

Old:
![image](https://user-images.githubusercontent.com/89415310/177424941-ae32ba03-2eac-4999-9a7e-d5421d1b4a9c.png)

Thoughts on this?  I personally prefer the ISO format in the old date time picker, but the type face and look of the new one looks pretty great.  If possible my choice would be the look of the new picker with the format from <= 2.3.0.

Airflow uses ISO for other dates in the program (`ds` etc.) so that's why I wanted to start this discussion.

Thanks!

**Comment by alex-astronomer** [comment]
@bbovenzi As promised

**Comment by bbovenzi** [comment]
I'd say that we should probably use ISO format since that has been the default across the app. It should be an easy change if you want to open a PR!

**Reply by JoranDox** [reply]
Any pointers as to where to start for this? I'm unreasonably annoyed by non-iso date formats :smile:

**Reply by bbovenzi** [reply]
I'd check out the date input [here](https://github.com/apache/airflow/blob/main/airflow/www/static/js/grid/FilterBar.jsx#L54)

**Reply by RNHTTR** [reply]
Not only because it's been the default, but it removes all ambiguity, and makes it much easier (IMO) to view the order of things when they're ordered chronologically.

"5/4/2022" means something different in the US that it does in other places. ISO format is consistent no matter where you're from.

**Reply by JoranDox** [reply]
So after googling around a bit I realised that react isn't choosing the list of possible form input types, they're just html forms, and there's only  datetime-local / date (also non-iso) / time. It seems to go a lot deeper than just changing a type somewhere, internally it's saved as an iso date, but it's the browser showing it as a localised date.

I played around a bit with this codepen (not mine) https://codepen.io/WebReflection/pen/MoYoNy and (see also https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#value) it seems like the core issue is that my locale is apparently near-impossible to change to use iso format dates instead of silly dd/mm/yyyy.

That said, we could just ditch the date picker and request text input, but that's not a great solution.

**Reply by bbovenzi** [reply]
Oh you're right. Maybe we need to use a specific datetime picker that we can control more: https://github.com/chakra-ui/chakra-ui/issues/580
